Implemented orders (shopping cart) logic.

GianSz opened the pull request:
- Created Order controller, Order model, OrderItem model and its corresponding migrations. 
- 'routes.php' file has been modified by adding the necessary endpoints for the app to work. 
- 'constants.php' file has been modified by adding a constant called 'SHIPPING_VALUE'.
- Views corresponding to the orders have been created.

GianSz committed:
Implemented orders (shopping cart) logic.
Created Order controller, Order model, OrderItem model and its corresponding migrations.
'routes.php' file has been modified by adding the necessary endpoints for the app to work.
'constants.php' file has been modified by adding a constant called 'SHIPPING_VALUE'.
Views corresponding to the orders have been created.

Andres-Alvarez-V commented on motohub_laravel/resources/views/order/index.blade.php:
Entre la linea 30 y 54 usar los getters respectivos es decir, los getters de motorcycle dado que los datos pueden luego llevar un formato distinto que cambiaria en el get respectivo

Andres-Alvarez-V reviewed (changes requested):
Has los cambios requeridos, luego el PR. Todo deberia estar listo luego de esa correcion

FelipeArredond commented on motohub_laravel/app/Http/Controllers/OrderController.php:
```diff
@@ -0,0 +1,122 @@
+        Order::validateSave($request);
+
+        $user_id = auth()->user()->id;
+        $user = User::findOrFail($user_id);
```
$userId

FelipeArredond replied:
$user = User::findOrFail($userId)

FelipeArredond commented on motohub_laravel/app/Http/Controllers/OrderController.php:
```diff
@@ -0,0 +1,122 @@
+
+        Order::validateSave($request);
+
+        $user_id = auth()->user()->id;
```
$userId

FelipeArredond commented on motohub_laravel/app/Http/Controllers/OrderController.php:
```diff
@@ -0,0 +1,122 @@
+        ]);
+
+        $order = Order::create([
+            'user_id' => $user_id,
```
'user_id' => $userId,

FelipeArredond commented on motohub_laravel/app/Models/OrderItem.php:
```diff
@@ -0,0 +1,61 @@
+        return $this->attributes['id'];
+    }
+
+    public function setId(int $id): void
```
Me parece innecesario un setter para el order_id ya que este es gestionado automaticamente por eloquent y no veo algun caso en el que sea necesario cambiar este id

FelipeArredond commented on motohub_laravel/resources/views/order/save.blade.php:
```diff
@@ -0,0 +1,7 @@
+@extends('layouts.app')
+@section('title', 'Orden')
```
El titulo debe ser dinamico a traves de lang para el idioma de la app ( feature a realizar a futuro )

FelipeArredond commented on motohub_laravel/app/Http/Controllers/OrderController.php:
Se debe mantener el camel case para las variables que se definan y hacer uso de snake case netamente al momento de referirse a la tabla de la bd.

Revisar comentarios entre la line 85 y 99

GianSz committed:
Merge branch 'develop' into sanchez

GianSz committed:
Requested changes made.
Added key => value pairs added into messages.php to manage the text in the orders view dinamically.
